Users/vapisar/dri readiness

vapisar opened the pull request:
This patch contains several fixes for CachePurge around cache purge statuses and APIs:
- tests were fixed to support added features and fixes (like auth, new object formats, statuses);
- cache purge statuses are updated and propagated to the API;
- API cleanups (obsolete parameters were removed, plugins can now be enabled/disabled, unnecessary methods removed from Swagger).

P.S. These branch and PR are based on a previous [PR](https://github.com/microsoft/cachepurge/pull/2)

vapisar committed:
Switching to anonymous access for Swagger

vapisar committed:
Remove auth from all Azure Functions

vapisar committed:
- Adding an example for POST body;
- Using hostnames to resolve relative URLs

vapisar committed:
- Adding a package for Azure Functions back to the project with Azure Functions;
- plus trivial documentation

vapisar committed:
Trivial, moving Guid to a parameter of the Function

vapisar committed:
Trivial, function documentation

vapisar committed:
Updating tests

vapisar committed:
Adding user names to logs

vapisar committed:
Patching tests for principal calls

vapisar committed:
Patching tests to not fail with null loggers

vapisar committed:
Minor: do not fail because of logging in case (for some unknown reason) ClaimsPrincipal doesn't have all the data in it

vapisar committed:
Adding metrics that we were tracking before

vapisar committed:
Swashbuckle shows ClaimsPrincipals as a textarea that gets submitted even for GET requests (and obviously doesn't work). We'll get users from HttpRequests then

vapisar committed:
Adding functions to check user groups and claims in Azure AD

vapisar committed:
Checking roles too

vapisar committed:
Checking claims and app roles

vapisar committed:
Trivial, typo fixed

vapisar committed:
Serializing claims and their properties

vapisar committed:
Adding authorization checks to important functions in Cache Purge

vapisar committed:
Altering permissions for CachePurge methods

vapisar committed:
Removing Function auth to simplify things for DRIs

vapisar committed:
Making sure that tests pass in pipelines

vapisar committed:
Auth checker function added

vapisar committed:
Hiding Swagger functions behind auth checks

vapisar committed:
Small improvements for the API:
- removing unused fields (like emails);
- enabling/disabling different CDN plugins;
- fixing exceptions when querying non-existing entities (like statuses for a request that wasn't submitted);
- removing unused settings for CDN credentials;
- removing functions related to user auth from Swagger UI;
- marking requests as completed correctly;
- preparing functions for deployment and testing, etc.

vapisar committed:
Validators for automatic tests

vapisar committed:
Adding more tests & fixing existing tests (handling and testing different testing and prod environments in auth, handling and testing old and new formats for partners, fixing tests locally and remotely)

vapisar committed:
Fixing the case when older formats are not showing up in the API

vapisar committed:
Prettifying partner results a little bit

vapisar committed:
Minor: improving local debugging + improving status messages

vapisar committed:
Minor: fixing tests and using the same format everywhere

vapisar committed:
Trivial, removing useless settings

shirleensharma commented on CachePurgeLibrary/RequestModels/CdnConfiguration.cs:
```diff
@@ -11,24 +11,29 @@ namespace CachePurgeLibrary
 
-        public IDictionary<string, string> CdnWithCredentials { get; set; }
+        // Credentials are not used for now anyway
+        // public IDictionary<string, string> CdnWithCredentials { get; set; }
```
Delete commented out code

vapisar replied:
You're absolutely right, removed those everywhere.

shirleensharma commented on MultiCdnApi/src/Functions/CacheFunctions.cs:
```diff
@@ -60,33 +72,45 @@ public class CacheFunctions
                 {
-                    var cdnWithCredentials = partnerCdnConfiguration.CdnWithCredentials;
-                    foreach (var cdnWithCredential in cdnWithCredentials)
+                    var cdn = Enum.Parse<CDN>(pluginName);
```
TryParse

vapisar committed:
Trivial, removing commented out code

vapisar committed:
Minor fixes:
- Addressing after-review comments from the PR;
- Removing an unused hostname field;
- Improving documentation a little bit.

vapisar committed:
Trivial, fixing tests

vapisar committed:
Trivial, updated documentation

vapisar committed:
Trivial, checking the hostname for null and empty values too
